File: tdc3/data/tokenizer/js/util/ioUtil.js
(function() {

    const fs = require("fs");

    function readFiles(filesArg) {
        let files;
        if (filesArg.length == 1 && filesArg[0].endsWith(".txt")) {
            files = fs.readFileSync(filesArg[0], { encoding: "utf8" }).split(/\r?\n/);
            files = files.filter(f => f.length > 0);
        } else {
            files = filesArg;
        }
        return files;
    }

    module.exports.readFiles = readFiles;

})();

File: tdc3/data/embedder/util/IOUtil.py
def read_files(files_arg):
    if len(files_arg) == 1 and files_arg[0].endswith(".txt"):
        files = []
        with open(files_arg[0]) as fp:
            for line in fp.readlines():
                files.append(line.rstrip())
    else:
        files = files_arg
    return files